packages/apps/job-launcher/server/src/modules/job/job.service.ts

```diff
@@ -633,65 +634,62 @@ export class JobService {
 
-    const { escrowAddress } = jobEntity;
-    if (escrowAddress) {
-      const signer = this.web3Service.getSigner(jobEntity.chainId);
+    if (jobEntity.escrowAddress) {
+      const { amountRefunded } = await this.processEscrowCancellation(jobEntity);
+      await this.paymentService.createRefundPayment({ refundAmount: Number(ethers.utils.formatEther(amountRefunded)), userId: jobEntity.userId, jobId: jobEntity.id });
+    } else {
+      await this.paymentService.createRefundPayment({ refundAmount: jobEntity.fundAmount, userId: jobEntity.userId, jobId: jobEntity.id });
+    }
+
+    jobEntity.status = JobStatus.CANCELED;
+    await jobEntity.save();
+
+    this.notifyWebhook(jobEntity);
+
+    return true;
+  }
+
+  public async processEscrowCancellation(jobEntity: JobEntity): Promise<EscrowCancelDto> {
+      const { chainId, escrowAddress } = jobEntity;
+
+      const signer = this.web3Service.getSigner(chainId);
       const escrowClient = await EscrowClient.build(signer);
 
       const escrowStatus = await escrowClient.getStatus(escrowAddress);
-      if (
-        escrowStatus === EscrowStatus.Complete ||
-        escrowStatus === EscrowStatus.Paid
-      ) {
-        this.logger.log(ErrorEscrow.InvalidStatusCancellation, JobService.name);
-        throw new BadRequestException(ErrorEscrow.InvalidStatusCancellation);
+      if (escrowStatus === EscrowStatus.Complete || escrowStatus === EscrowStatus.Paid || escrowStatus === EscrowStatus.Cancelled) {
+          this.logger.log(ErrorEscrow.InvalidStatusCancellation, JobService.name);
+          throw new BadRequestException(ErrorEscrow.InvalidStatusCancellation);
       }
 
       const balance = await escrowClient.getBalance(escrowAddress);
       if (balance.eq(0)) {
-        this.logger.log(
-          ErrorEscrow.InvalidBalanceCancellation,
-          JobService.name,
-        );
-        throw new BadRequestException(ErrorEscrow.InvalidBalanceCancellation);
+          this.logger.log(ErrorEscrow.InvalidBalanceCancellation, JobService.name);
+          throw new BadRequestException(ErrorEscrow.InvalidBalanceCancellation);
       }
 
-      await escrowClient.cancel(escrowAddress);
+      return escrowClient.cancel(escrowAddress);
+  }
 
-      const manifest = await this.getManifest(jobEntity.manifestUrl);
-      if (
-        (manifest as FortuneManifestDto).requestType === JobRequestType.FORTUNE
-      ) {
-        await this.sendWebhook(
-          this.configService.get<string>(
-            ConfigNames.FORTUNE_EXCHANGE_ORACLE_WEBHOOK_URL,
-          )!,
-          {
-            escrowAddress,
-            chainId: jobEntity.chainId,
-            eventType: EventType.ESCROW_CANCELED,
-          },
-        );
-      } else {
-        await this.sendWebhook(
-          this.configService.get<string>(
-            ConfigNames.CVAT_EXCHANGE_ORACLE_WEBHOOK_URL,
-          )!,
-          {
-            escrowAddress,
+  public async notifyWebhook(jobEntity: JobEntity) {
```

Comment:
This notifyWebhook cannot contain the code to set the payment to FAILED. In any case I think we don't need to set the payment as FAILED since the payment was actuially successfull and we are creating the REFUND.
Also does not make sense to have a notify webhook specific for cancellation, lets just use the generic send webhook, which can be still more generic like this:
```
public async sendWebhook(
    webhookUrl: string,
    webhookData: any,
  )
```

Reply:
Haven't checked the content of this method, it should work differently. I'll fix it.